lib/core/storage/clientAdapter.ts

```diff
@@ -19,36 +19,51 @@
  * limitations under the License.
  */
 
-"use strict";
-
-const Elasticsearch = require("../../service/storage/elasticsearch");
-const { IndexCache } = require("./indexCache");
-const { isPlainObject } = require("../../util/safeObject");
-const kerror = require("../../kerror");
-const { Mutex } = require("../../util/mutex");
+import { Elasticsearch } from "../../service/storage/elasticsearch";
+import { IndexCache } from "./indexCache";
+import { isPlainObject } from "../../util/safeObject";
+import * as kerror from "../../kerror";
+import { Mutex } from "../../util/mutex";
+import { VirtualIndex } from "../../service/storage/virtualIndex";
 
 const servicesError = kerror.wrap("services", "storage");
 
 /**
  * Storage client adapter to perform validation on index/collection existence
  * and to maintain the index/collection cache.
  */
-class ClientAdapter {
+export class ClientAdapter {
+  client: Elasticsearch;
+  cache: IndexCache;
+  scope: string;
+  virtualIndex: VirtualIndex;
+
+  static getMutex() {
```

Comment:
I will re state what I've said before
> I dont think this a good design to create a static function with a generic  name to make a mutex,
mutexes have specifics names that are used to lock specific actions / ressources but having something like `ClientAdapter.getMutex()` might lead someone to think that if a mutex is needed in one of the ClientAdapter function they should use this method to acquire the mutex, which is wrong because the mutex will lock a ressource with the same name (`loadMappings`) which might be totally unrelated to mappings loading and impact an other node that might be starting and checking if one of the node is finishing or has already finished the initialization sequence.

You can create a generic `getMutex` which take mutex name and mutex options this way you keep it clear when acquiring mutexes and you can mock it without difficulty

**example**
```ts
  static getMutex(mutexName: string, options: MutexOptions) {
```